### ISSUE TITLE
gemini-exp-1121

### ISSUE THREAD

**Issue by simonw** [simonw/llm-gemini#26]
https://twitter.com/officiallogank/status/1859667244688736419?s=46

> Say hello to gemini-exp-1121! Our latest experimental gemini model, with:
> 
> - significant gains on coding performance
> - stronger reasoning capabilities
> - improved visual understanding
> 
> Available on Google AI Studio and the Gemini API right now

**Comment by simonw** [comment]
```bash
llm -m gemini-exp-1121 'Generate an SVG of a pelican riding a bicycle'
```

> Okay, here's the SVG code for a pelican riding a bicycle. I've tried to make it somewhat whimsical and simple:
> 
> ```xml
> <svg width="300" height="250" viewBox="0 0 300 250" fill="none" xmlns="http://www.w3.org/2000/svg">
>   
>   <path d="M50 200H250M80 200V150C80 130 100 110 120 110H220C240 110 260 130 260 150V200M120 110L100 80M220 110L240 80M160 160C180 160 180 180 160 180C140 180 140 160 160 160Z" stroke="black" stroke-width="3"/>
>   <circle cx="100" cy="210" r="20" fill="black"/>
>   <circle cx="240" cy="210" r="20" fill="black"/>
> 
>   
>   <path d="M150 100C150 80 170 60 200 60C230 60 250 80 250 100C250 120 230 140 200 140C170 140 150 120 150 100Z" fill="#FFD700" stroke="black" stroke-width="2"/>
> 
>   
>   <path d="M130 100C110 100 90 120 90 140C90 160 110 180 130 180C150 180 170 160 170 140C170 120 150 100 130 100Z" fill="#FFD700" stroke="black" stroke-width="2"/>
> 
>   
>   <path d="M80 140C70 140 60 150 60 160C60 170 70 180 80 180L100 140L80 140Z" fill="#FFA07A" stroke="black" stroke-width="2"/>
> 
>   
>   <circle cx="120" cy="130" r="5" fill="black"/>
> 
>   
>   <path d="M170 90L140 60L170 40" stroke="black" stroke-width="2"/>
>   <path d="M230 90L260 60L230 40" stroke="black" stroke-width="2"/>
> 
>   
>   <path d="M180 140L180 160M220 140L220 160" stroke="black" stroke-width="2" stroke-linecap="round"/>
> 
> </svg>
> ```
> 
> **Explanation:**
> 
> *   **Structure:** The SVG is organized into sections for the bicycle and the pelican (body, head, beak, eye, wings, feet).
> *   **Shapes:** Basic shapes like `path` (for lines and curves), and `circle` are used to construct the image.
> *   **Attributes:**
>     *   `width`, `height`: Dimensions of the SVG canvas.
>     *   `viewBox`: Defines the coordinate system for scaling.
>     *   `fill`: Sets the fill color of a shape.
>     *   `stroke`: Sets the outline color.
>     *   `stroke-width`: Sets the thickness of the outline.
>     *   `cx`, `cy`: Center coordinates for a circle.
>     *   `r`: Radius of a circle.
>     *   `stroke-linecap`: Sets the style of line endings (rounded here for the feet).
> *   **Colors:**
>     *   `black`: Used for outlines and some details.
>     *   `#FFD700`: A gold color for the pelican's body and head.
>     *   `#FFA07A`: A light salmon color for the beak.
> *   **Paths:** The `d` attribute of the `path` elements defines the shape using a series of commands like `M` (moveto), `L` (lineto), `C` (curveto), `Z` (closepath).
> 
> **How to Use:**
> 
> 1. **Save:** Save the code above as a `.svg` file (e.g., `pelican_bike.svg`).
> 2. **Open:** You can open this file directly in a web browser, or you can embed it into an HTML file using the `<img>` tag:
> 
>     ```html
>     <!DOCTYPE html>
>     <html>
>     <head>
>       <title>Pelican on a Bicycle</title>
>     </head>
>     <body>
>       <img src="pelican_bike.svg" alt="Pelican riding a bicycle">
>     </body>
>     </html>
>     ```
> 
> This will display the image of a pelican humorously perched on a bicycle. You can adjust the colors, sizes, and shapes in the SVG code to customize the image further.

![gem](https://github.com/user-attachments/assets/21f3a02c-26fd-490c-9ec5-68d3de4ecbc6)

**Comment by simonw** [comment]
I'm releasing this as 0.4.2 rather than as 0.5 because I've already shipped alphas of 0.5 that have features I'm not yet ready to ship in a non-alpha release, because they depend on LLM alphas.

Not ideal (should be a major bump) but I'll live with it.

**Comment by simonw** [comment]
Ran that bad pelican in reverse:
```bash
llm -m gemini-exp-1121 describe -a /tmp/bad-pelican.png 
```

> The image shows a simple, stylized drawing of an insect, possibly a bee or an ant, on a vehicle. The insect is composed of a large yellow circle for the body and a smaller yellow circle for the head. It has a black dot for an eye, a small orange oval for a beak or mouth, and thin black lines for antennae and legs. The insect is positioned on top of a simple black and white vehicle with two black wheels. The drawing is abstract and geometric, using basic shapes and a limited color palette of black, white, yellow, and orange.
